app/routes/products.$categorySlug/route.tsx

```diff
@@ -113,3 +119,31 @@ export default function ProductsPage() {
         </div>
     );
 }
+
+export function ErrorBoundary() {
+    const error = useRouteError();
+    const navigate = useNavigate();
+
+    if (isRouteErrorResponse(error)) {
+        let title: string;
+        let message: string | undefined;
+        if (error.data.code === EcomApiErrorCodes.CategoryNotFound) {
+            title = 'Category Not Found';
+            message = "Unfortunately a category you trying to open doesn't exist";
+        } else {
+            title = 'Failed to load category products';
```

Comment:
I think a generic "Error" would be better, with the message explaining the details.